packages/schema/src/language-server/zmodel-workspace-manager.ts

```diff
@@ -1,21 +1,115 @@
+        await this.loadAdditionalDocuments(folders, collector);
+        await Promise.all(
+            folders
+                .map((wf) => [wf, this.getRootFolder(wf)] as [WorkspaceFolder, URI])
+                .map(async (entry) => this.traverseFolder(...entry, fileExtensions, collector))
+        );
+
+        // find plugin models
+        documents.forEach((doc) => {
+            const parsed = doc.parseResult.value as Model;
+            parsed.declarations.forEach((decl) => {
+                if (isPlugin(decl)) {
+                    const providerField = decl.fields.find((f) => f.name === 'provider');
+                    if (providerField) {
+                        const provider = getLiteral<string>(providerField.value);
+                        if (provider) {
+                            this.pluginModels.add(provider);
+                        }
+                    }
+                }
+            });
+        });
+
+        console.log(`Used plugin documents: ${Array.from(this.pluginModels)}`);
+
+        await Promise.all(
+            folders
+                .map((wf) => [wf, this.getRootFolder(wf)] as [WorkspaceFolder, URI])
+                .map(async (entry) => this.loadPluginModels(...entry, new Set(this.pluginModels), collector))
+        );
+
+        // Only after creating all documents do we check whether we need to cancel the initialization
+        // The document builder will later pick up on all unprocessed documents
+        await interruptAndCheck(cancelToken);
+        await this.documentBuilder.build(documents, undefined, cancelToken);
+    }
+
+    protected async loadPluginModels(
+        workspaceFolder: WorkspaceFolder,
+        folderPath: URI,
+        pluginModels: Set<string>,
+        collector: (document: LangiumDocument) => void
+    ): Promise<void> {
+        const content = await this.fileSystemProvider.readDirectory(folderPath);
+
+        for (const entry of content) {
+            if (entry.isDirectory) {
+                const name = Utils.basename(entry.uri);
+                if (name === 'node_modules') {
+                    for (const plugin of pluginModels) {
+                        const path = Utils.joinPath(entry.uri, plugin, PLUGIN_MODULE_NAME);
+                        try {
+                            this.fileSystemProvider.readFileSync(path);
+                            const document = this.langiumDocuments.getOrCreateDocument(path);
+                            collector(document);
+                            console.log(`Adding plugin document from ${path}`);
+
+                            pluginModels.delete(plugin);
+                            // early exit if all plugins are loaded
+                            if (pluginModels.size === 0) {
+                                return;
+                            }
+                        } catch {
+                            //no-op
```

Comment:
Shall we add a warning log here?

Reply:
There might be multiple node_modules locations,  the warning will be shown eventually after traversing all the folders but not found.